velox/common/memory/Memory.h

```diff
@@ -140,32 +148,56 @@ class MemoryManager final : public IMemoryManager {
 
   uint16_t alignment() const final;
 
-  MemoryPool& getRoot() const final;
+  std::shared_ptr<MemoryPool> getPool(
+      const std::string& name = "",
+      MemoryPool::Kind kind = MemoryPool::Kind::kAggregate,
+      int64_t cap = kMaxMemory) final;
 
-  std::shared_ptr<MemoryPool> getChild(int64_t cap = kMaxMemory) final;
+  MemoryPool& deprecatedGetPool() final;
 
   int64_t getTotalBytes() const final;
 
   bool reserve(int64_t size) final;
   void release(int64_t size) final;
 
+  /// Returns the number of alive user memory pools including both root and leaf
+  /// memory pools.
+  ///
+  /// NOTE: this doesn't count the memory manager's internal default root and
+  /// leaf memory pools.
+  size_t numUserPools() const;
```

Comment:
Any particular reason to use 'User' in the name of this method. I'd expect numPools() name to match getPool().

Reply:
The default memory pool is not counted. Yeah, numPools are better. Changed.